File: harmoniq_backend/src/main/java/dev/anirban/harmoniq_backend/dto/thread/ThreadDto.java
package dev.anirban.harmoniq_backend.dto.thread;

import dev.anirban.harmoniq_backend.dto.user.UserDto;
import lombok.*;

import java.util.List;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ThreadDto {
    private String id;
    private String description;
    private UserDto createdBy;
    private List<String> tags;
    private Integer totalLikes;
    private List<String> likedByUserIds;
    private Integer totalComments;
}

File: harmoniq_frontend/src/app/shared/Models/thread/ThreadDto.ts
import { UserDto } from '../user/UserDto';

export interface ThreadDto {
  id: string;
  description: string;
  tags: string[];
  createdBy: UserDto;
  totalLikes: number;
  likedByUserIds: string[];
  isLikedByCurrentUser: boolean;
  totalComments: number;
}
